frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -196,7 +197,7 @@ export const fetchDashboard = createAsyncThunk(
         );
         result = {
           ...result,
-          id: dashId,
+          id: IS_EMBED_PREVIEW ? result.id : dashId,
```

Comment:
This is essentially the fix. Previously `dashId` in this case (`dashboardType === "embed"`) would be a UUID, and that makes calling the parameter values API with the dashboard as a UUID

But that means we'd be calling `/api/dashboard/:uuid` which is wrong, we're supposed to call `/api/dashboard/:dashboard-id` instead. The thing is that in embed preview we don't call `/api/embed/*` endpoints because it's possible for the resource to not have been embedded yet, calling `/api/embed/*` in that case will not return anything.

You can see that we have a logic that changes the dashboard and question parameter values endpoints to have `/api/embed` prefix for embedding, but we don't do that in embed preview
https://github.com/metabase/metabase/blob/e2f07d813d0c60ad7131d6c3abb9e4e838757a76/frontend/src/metabase/services.js#L531-L540